### ISSUE TITLE
Rework DSStore from perl to a Gradle task

### ISSUE THREAD

**Issue by vlsi** [mbeddr/mps-gradle-plugin#71]
The current code is somewhat hard to follow, and it is not clear why `arch -32` is needed:

https://github.com/mbeddr/mps-gradle-plugin/blob/24e832c4d4acc272b70663c9b64c87cc1c12cb5e/src/main/resources/de/itemis/mps/gradle/mpsdmg.sh#L56

What if `DS_Store` generation was written in Java-based code?
For instance: https://formats.kaitai.io/ds_store/java.html

**Comment by vlsi** [comment]
I see IntelliJ IDEA does not use `arch -32`: https://github.com/JetBrains/intellij-community/blob/c838dd4efbd00629f2a5874f336a324a04a8c141/platform/build-scripts/tools/mac/scripts/makedmg.sh#L81

@sergej-koscejev, do you by chance remember why `arch -32` is used for `perl5.18 mpsdmg.pl` execution in `mps-gradle-plugin`?

**Comment by vlsi** [comment]
An alternative approach migh be to reuse IDEA's build scripts.

For instance, they are still improved: https://github.com/JetBrains/intellij-community/commit/c838dd4efbd00629f2a5874f336a324a04a8c141

**Comment by sergej-koscejev** [comment]
Re `arch -32`: I copied the scripts over from mbeddr.core, I have no idea why it was used there. Here's relevant Git Blame for mbeddr.core: https://github.com/mbeddr/mbeddr.core/blame/a44cf0012dd573a403a5c340d65e02cebda2441d/buildUtil/mpsdmg.sh#L45

**Comment by sergej-koscejev** [comment]
@vlsi is there a particular feature that's missing or not working? What motivated you to focus on this code?

**Comment by vlsi** [comment]
@sergej-koscejev , AFAIK one of the issues is the current Perl script fails at GitHub Actions in macOS (e.g. https://github.com/mbeddr/mps-gradle-plugin/pull/70/checks?check_run_id=902243688 )
